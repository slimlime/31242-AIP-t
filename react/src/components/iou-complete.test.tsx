import React from "react";
import { shallow, mount } from "enzyme";
import IouComplete from "./iou-complete";

// simple, static component
describe("<IouComplete />", () => {
  global.URL.createObjectURL = jest.fn();
  it("should render correctly", () => {
    const wrapper = shallow(
      <IouComplete
        id="1"
        is_completed={true}
        author="James"
        completed_by="Kevin"
        claimed_time="01/06/2020"
        created_time="01/01/2020"
        rewards={[
          {
            id: "1",
            giver: { username: "James", display_name: "James" },
            item: { id: "201", display_name: "Coffee" },
          },
        ]}
        details="Clean the fridge"
        iouType={0}
        refreshTable={() => {
          return "substitute test function";
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a disabled complete button on a owing IOU on render", () => {
    const wrapper = mount(
      <IouComplete
        id="1"
        is_completed={false}
        author="James"
        completed_by=""
        claimed_time={null}
        created_time="01/01/2020"
        rewards={[
          {
            id: "1",
            giver: { username: "James", display_name: "James" },
            item: { id: "201", display_name: "Coffee" },
          },
        ]}
        details="Clean the fridge"
        iouType={1}
        refreshTable={() => {
          return "substitute test function";
        }}
      />
    );
    wrapper.find("div.cursorPointer").simulate("click");
    expect(wrapper.state("completeIOU")).toBe(true);
    expect(wrapper.find("button#createRequest").props().disabled).toBe(true);
  });

  it("should have an enabled complete button on a owed IOU on render", () => {
    const wrapper = mount(
      <IouComplete
        id="1"
        is_completed={false}
        author="James"
        completed_by=""
        claimed_time={null}
        created_time="01/01/2020"
        rewards={[
          {
            id: "1",
            giver: { username: "James", display_name: "James" },
            item: { id: "201", display_name: "Coffee" },
          },
        ]}
        details="Clean the fridge"
        iouType={0}
        refreshTable={() => {
          return "substitute test function";
        }}
      />
    );
    wrapper.find("div.cursorPointer").simulate("click");
    expect(wrapper.state("completeIOU")).toBe(true);
    expect(wrapper.find("button#createRequest").props().disabled).toBe(false);
  });

  it("should have a disabled complete button for requests on render", () => {
    const wrapper = mount(
      <IouComplete
        id="1"
        is_completed={false}
        author="James"
        completed_by=""
        claimed_time={null}
        created_time="01/01/2020"
        rewards={[
          {
            id: "1",
            giver: { username: "James", display_name: "James" },
            item: { id: "201", display_name: "Coffee" },
          },
        ]}
        details="Clean the fridge"
        iouType={2}
        refreshTable={() => {
          return "substitute test function";
        }}
      />
    );
    wrapper.find("div.cursorPointer").simulate("click");
    expect(wrapper.state("completeIOU")).toBe(true);
    expect(wrapper.find("button#createRequest").props().disabled).toBe(true);
  });

  it("should have an enabled complete button for requests on image upload", () => {
    const wrapper = mount(
      <IouComplete
        id="1"
        is_completed={false}
        author="James"
        completed_by=""
        claimed_time={null}
        created_time="01/01/2020"
        rewards={[
          {
            id: "1",
            giver: { username: "James", display_name: "James" },
            item: { id: "201", display_name: "Coffee" },
          },
        ]}
        details="Clean the fridge"
        iouType={2}
        refreshTable={() => {
          return "substitute test function";
        }}
      />
    );
    wrapper.find("div.cursorPointer").simulate("click");
    expect(wrapper.state("completeIOU")).toBe(true);
    global.URL.createObjectURL = jest.fn((param: any) => "tempUrlString");
    wrapper.setState({ submittedProof: "Temp File Mockup" });
    expect(wrapper.find("button#createRequest").props().disabled).toBe(false);
  });

  it("should have an enabled complete button for owing IOU on image upload", () => {
    const wrapper = mount(
      <IouComplete
        id="1"
        is_completed={false}
        author="James"
        completed_by=""
        claimed_time={null}
        created_time="01/01/2020"
        rewards={[
          {
            id: "1",
            giver: { username: "James", display_name: "James" },
            item: { id: "201", display_name: "Coffee" },
          },
        ]}
        details="Clean the fridge"
        iouType={1}
        refreshTable={() => {
          return "substitute test function";
        }}
      />
    );
    wrapper.find("div.cursorPointer").simulate("click");
    expect(wrapper.state("completeIOU")).toBe(true);
    global.URL.createObjectURL = jest.fn((param: any) => "tempUrlString");
    wrapper.setState({ submittedProof: "Temp File Mockup" });
    expect(wrapper.find("button#createRequest").props().disabled).toBe(false);
  });
});
