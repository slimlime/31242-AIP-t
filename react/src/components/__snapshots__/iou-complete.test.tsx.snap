// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IouComplete /> should render correctly 1`] = `
<div
  id="completeRequestItem"
>
  <div
    className=""
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Checkbox))
      checked={true}
      checkedIcon={
        <UNDEFINED
          color="primary"
          id="completeIconSize"
        />
      }
      color="primary"
      disabled={true}
      icon={
        <UNDEFINED
          color="primary"
          id="completeIconSize"
        />
      }
    />
    <div
      id="IouTaskComplete"
    >
      <div>
        <WithStyles(ForwardRef(Typography))
          id="taskCompleter"
        >
          Kevin
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          id="taskTimeStamp"
        >
          01/06/2020
        </WithStyles(ForwardRef(Typography))>
      </div>
      <WithStyles(ForwardRef(Popover))
        anchorEl={null}
        anchorOrigin={
          Object {
            "horizontal": "center",
            "vertical": "bottom",
          }
        }
        id="popUpMargin"
        onClose={[Function]}
        open={false}
        transformOrigin={
          Object {
            "horizontal": "center",
            "vertical": "top",
          }
        }
      >
        <WithStyles(ForwardRef(Typography))
          id="taskDetailPopUp"
        >
          Full Display Name
          <WithStyles(ForwardRef(Divider))
            id="taskPopUpDivider"
          />
          Kevin
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Popover))>
    </div>
  </div>
  <WithStyles(ForwardRef(Dialog))
    TransitionComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "muiSupportAuto": true,
        "propTypes": Object {
          "children": [Function],
          "disableStrictModeCompat": [Function],
          "in": [Function],
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "style": [Function],
          "timeout": [Function],
        },
        "render": [Function],
      }
    }
    fullWidth={true}
    id="completeForm"
    maxWidth="xs"
    onClose={[Function]}
    open={false}
    scroll="body"
  >
    <WithStyles(ForwardRef(DialogTitle))
      disableTypography={true}
    >
      <WithStyles(ForwardRef(Typography))
        id="modalTitle"
        variant="h5"
      >
        IOU Proof Submission
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(DialogContent))
      className="content"
      id="completeDetailsTable"
    >
      <WithStyles(ForwardRef(DialogContentText))
        variant="h6"
      >
        IOU Details
      </WithStyles(ForwardRef(DialogContentText))>
      <WithStyles(ForwardRef(TableContainer))
        id="infoTable"
      >
        <WithStyles(ForwardRef(Table))>
          <WithStyles(ForwardRef(TableBody))>
            <WithStyles(ForwardRef(TableRow))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                className="infoTableCell"
                id="infoTableTitle"
              >
                IOU
                 created by:
              </WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                className="infoTableCell"
                id="infoTableContent"
              >
                James
              </WithStyles(ForwardRef(TableCell))>
            </WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableRow))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableTitle"
              >
                IOU
                 Created on:
              </WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableContent"
              >
                01/01/2020
              </WithStyles(ForwardRef(TableCell))>
            </WithStyles(ForwardRef(TableRow))>
            <WithStyles(ForwardRef(TableRow))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableTitle"
              >
                Will be completed by:
              </WithStyles(ForwardRef(TableCell))>
              <WithStyles(ForwardRef(TableCell))
                align="center"
                id="infoTableContent"
              >
                Kevin
              </WithStyles(ForwardRef(TableCell))>
            </WithStyles(ForwardRef(TableRow))>
          </WithStyles(ForwardRef(TableBody))>
        </WithStyles(ForwardRef(Table))>
      </WithStyles(ForwardRef(TableContainer))>
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(DialogActions))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        size="large"
      >
        Nevermind
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        autoFocus={true}
        color="primary"
        disabled={false}
        id="createRequest"
        onClick={[Function]}
        size="large"
      >
        Submit & Complete
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
  <WithStyles(ForwardRef(Snackbar))
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    autoHideDuration={5000}
    message=""
    onClose={[Function]}
    open={false}
  />
</div>
`;
